### ISSUE TITLE
Validate/Touch array fields themselves as well as the items in the array

### ISSUE THREAD

**Issue by timc13** [jaredpalmer/formik#442]
## Feature

Allow `errors` & `touched` to work on arrays & objects, not just it's items (array) or children (object).

## Current Behavior

This is not as relevant in objects, because the form validation for an object does not make much sense, but an array field, we often want to validate `min/max` of an array.  

There's no way to do that right now without breaking a lot of behavior.  If I touch an array field, it will set it to `true` which breaks all `FieldArray` functionality because it expects an array - not a boolean value.

## Desired Behavior
Have a convention for `errors` and `touched` to store validation and touching of the object/array itself.
## Suggested Solutions
I'm not sure if this is totally possible, but it seems when setting `errors` or `touched` we would have to know of the structure of all the values - is that possible? If so, then we can do something like
append a `{}` or `[]` to the field name to store the data.

Example `errors`:
```
{
  foo: [
    0: "foo[0] is required.",
    1: "foo[1] is required.",
  ],
  foo[]: "foo is required to have at least 1 item.",
  bar : {
    child1: "Required",
    child2: "Required",
  },
  bar{}: "Required",
}
```
Example `touched`:
```
{
  foo: [
    0: true,
    1: true
  ],
  foo[]: true,
  bar : {
    child1: true,
    child2: true,
  },
  bar{}: true,
}
```
## Info
 

- Formik Version:  Next
- OS: ALL
- Node Version: ALL
- Package Manager and version: ALL

**Comment by jaredpalmer** [comment]
Yeah I thought about this—have a special key for array errors—when coding field array. Def something to consider about for v1

**Comment by timc13** [comment]
I believe the only way to reliably know whether a field is an array or an object is via `initialValues` or a Yup schema.  Perhaps this feature can only work when either one of those is provided.

**Comment by timc13** [comment]
this seems like a mistake... @jaredpalmer ?

**Comment by timc13** [comment]
@jaredpalmer is this being tracked elsewhere?...or is this no longer an enhancement that is planned
